18-hlt-intro.md

```diff
@@ -7,129 +7,233 @@ minutes: 45
-print report.decision()
+from Configurables import Moore
+# Define settings
+Moore().ThresholdSettings = "Physics_pp_2017"
+Moore().DataType = "2017"
+Moore().UseTCK = False
+Moore().RemoveInputHltRawBanks = True
+Moore().Split = ''
+# A bit more output
+Moore().EnableTimer = True
+Moore().OutputLevel = 3
+# Input data
+from PRConfig import TestFileDB
+TestFileDB.test_file_db["2016NB_25ns_L0Filt0x1609"].run(configurable=Moore())
+# Set the right L0 TCK
+from Configurables import HltConf
+HltConf().setProp("L0TCK", '0x1609')
+Moore().EvtMax = 1000
+# Remove a line which accepts every event when run on this sample
+HltConf().RemoveHlt1Lines = ["Hlt1MBNoBias"]
 ~~~
 
-The HLT1 selections that are most efficient for hadronic charm and beauty decays
-in Run 2 are called Hlt1TrackMVA and Hlt1TwoTrackMVA. Use the advance function
-to find an event that was accepted by either of these trigger selections.
+Try to run it with
+```
+$ lb-run Moore/latest gaudirun.py runMoore.py | tee log.txt
+```
 
-The DecReports only contains the decisions for each line, 1 or 0. The 
-candidates themselves are stored in the
-SelReports ("Hlt{1,2}/SelReports"). Get the HLT1 SelReports from the event store
-and retrieve the one for one of the TrackMVA selections using the selReport function,
-analogously to how the DecReport was retrieved above.
+There are two ways to run Moore, from ```ThresholdSettings``` and from ```TCK``` (Trigger Configuration Key).
+When you develop a trigger line, it is more convenient to run from ThresholdSettings. The TCK
+is used when running the trigger on the online farm or in MC productions as it uniquely defines the settings.
 
-The SelReports store candidates in a tree of sub-structures, which can be
-accessed using the `substructure` member funtion of a report. Any SelReport has at
-least one level of sub-structure. The sub-structure is internally stored in
-SmartRefs, which can be dereferenced using their "data" method. Let's have a look
-at a SelReport for a TrackMVA selection.
+> ## What is a TCK? {.callout}
+> The Trigger Configuration Key (TCK) stores the configuration of the HLT in a database.
+> All algorithms and their properties are defined in it.
+> The key is usually given as a hexadecimal number. The last 4 digits define the L0 TCK.
+> The first 4 digits define the HLT configuration. HLT1 TCKs start with 1, HLT2 TCKs start
+> with 2.
+
+The property split defines if HLT1, HLT2 or both are run. In the example above both are run.
+Change ```Moore().Split``` to ```'Hlt1'``` and rerun.
+To run HLT2 only, you have to change some settings and the input file has to be a file where
+HLT1 has run on:
 
 ~~~ {.python}
-reports = evt['Hlt1/SelReports']
-report = reports.selReport('Hlt1TrackMVADecision')
-print report
-report.substructure().size()
-report.substructure()[0].substructure().size()
-report.substructure()[0].substructure()[0]
-report.substructure()[0].substructure()[0].data()
+...
+Moore().RemoveInputHltRawBanks = False # Why?
```

Comment:
Ok it was intended as a question to the audience, because in Hlt2 you should not delete the Hlt1 raw banks. Just to get them thinking a bit how the data flow is. 
I will add ```print Moore()```.